tokio-postgres/src/client.rs

```diff
@@ -449,6 +451,15 @@ impl Client {
         copy_out::copy_out(self.inner(), statement).await
     }
 
+    /// Executes a CopyBoth query, returning a combined Stream+Sink type to read and write copy
+    /// data.
+    pub async fn copy_both_simple<T>(&self, query: &str) -> Result<CopyBothDuplex<T>, Error>
```

Comment:
After the replication stream, if the timeline is historical, Postgres will send a tuple as a response. So we actually need a function that returns something like `Result<(CopyBothDuplex<T>, Option<SimpleQueryMessage>), Error>` (or maybe `Result<(CopyBothDuplex<T>, Option<Vec<SimpleQueryMessage>>), Error>` in case other commands are added in the future which use `CopyBoth` and return a set).

It's actually very specific to `START_REPLICATION` (and even more specifically, to physical replication), so it might make sense to have a more specific name or at least clarify what it's expecting the command to do. Maybe something like `copy_both_simple_with_result()`?

Reply:
That's a good point, I'll take a look on how we can expose this to users, ideally in a generic way.

Reply:
In case you missed my other comment, there's a similar issue for `BASE_BACKUP`, except with `CopyOut` instead. That can be a separate PR, though.